udata/linkchecker/checker.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals, absolute_import
+
+from datetime import datetime, timedelta
+
+from flask import current_app
+
+from .backends import get as get_linkchecker
+
+
+def _get_check_keys(the_dict):
+    return {k: v for k, v in the_dict.iteritems() if k.startswith('check:')}
+
+
+def get_cache(resource):
+    '''Return a cached version from a resource's check if any and fresh'''
+    if resource.extras.get('check:date'):
+        cache_duration = current_app.config['LINKCHECKING_CACHE_DURATION']
+        if resource.extras['check:date'] <= datetime.now() \
```

Comment:
You might want to extract the expiration into an intermediary variable:
- prevent line break/split
- give meaning to the condition